### PR TITLE
Feature/keyvault msi integration test

### PR DESCRIPTION
**Pull request by saragluna**
## Summary
Add integration tests for Key Vault authenticating using MSI in Virtual Machine and App Service.

### CODE REVIEW

**Comment by saragluna** [comment]
Seems like Travis has [an issue with FTP uploading files](https://github.com/travis-ci/travis-ci/issues/9347).

**Comment by saragluna** [comment]
Possible answer https://blog.travis-ci.com/2018-07-23-the-tale-of-ftp-at-travis-ci

**Comment by Incarnation-p-lee** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- azure-spring-boot-tests/azure-spring-boot-test-cosmosdb/src/test/java/com/microsoft/azure/test/cosmosdb/CosmosDBIT.java  1
- azure-spring-boot-tests/azure-spring-boot-test-core/src/main/java/com/microsoft/azure/utils/SSHShell.java  6
- azure-spring-boot-tests/azure-spring-boot-test-core/src/main/java/com/microsoft/azure/mgmt/AppServiceTool.java  1
- azure-spring-boot-tests/azure-spring-boot-test-core/src/main/java/com/microsoft/azure/mgmt/VirtualMachineTool.java  1
- azure-spring-boot-tests/azure-spring-boot-test-keyvault/src/test/java/com/microsoft/azure/test/keyvault/KeyVaultIT.java  4
- azure-spring-boot-tests/azure-spring-boot-test-application/src/main/java/com/microsoft/azure/test/Application.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/Microsoft/azure-spring-boot/pullRequest?prid=4450600&bid=15044855)

File: azure-spring-boot-tests/azure-spring-boot-test-core/src/main/java/com/microsoft/azure/mgmt/AppServiceTool.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import com.microsoft.azure.management.appservice.RuntimeStack;
+import com.microsoft.azure.management.appservice.WebApp;
+import com.microsoft.azure.management.appservice.WebApps;
+import com.microsoft.azure.management.appservice.WebContainer;
+import com.microsoft.azure.management.graphrbac.BuiltInRole;
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
+import com.microsoft.azure.management.resources.fluentcore.utils.SdkContext;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.File;
```
Comment (Incarnation-p-lee) [inline azure-spring-boot-tests/azure-spring-boot-test-core/src/main/java/com/microsoft/azure/mgmt/AppServiceTool.java]: Codacy Issue found: [Avoid unused imports such as 'java.io.File'](https://app.codacy.com/app/Microsoft/azure-spring-boot/pullRequest?prid=4450600)

**Comment by Incarnation-p-lee** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- azure-spring-boot-tests/azure-spring-boot-test-core/src/main/java/com/microsoft/azure/utils/SSHShell.java  6
- azure-spring-boot-tests/azure-spring-boot-test-core/src/main/java/com/microsoft/azure/mgmt/AppServiceTool.java  1
- azure-spring-boot-tests/azure-spring-boot-test-cosmosdb/src/test/java/com/microsoft/azure/test/cosmosdb/CosmosDBIT.java  1
- azure-spring-boot-tests/azure-spring-boot-test-core/src/main/java/com/microsoft/azure/mgmt/VirtualMachineTool.java  1
- azure-spring-boot-tests/azure-spring-boot-test-application/src/main/java/com/microsoft/azure/test/Application.java  2
- azure-spring-boot-tests/azure-spring-boot-test-keyvault/src/test/java/com/microsoft/azure/test/keyvault/KeyVaultIT.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/Microsoft/azure-spring-boot/pullRequest?prid=4450600&bid=15044855)

**Comment by Incarnation-p-lee** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- azure-spring-boot-tests/azure-spring-boot-test-core/src/main/java/com/microsoft/azure/mgmt/AppServiceTool.java  1
- azure-spring-boot-tests/azure-spring-boot-test-core/src/main/java/com/microsoft/azure/mgmt/VirtualMachineTool.java  1
- azure-spring-boot-tests/azure-spring-boot-test-keyvault/src/test/java/com/microsoft/azure/test/keyvault/KeyVaultIT.java  4
- azure-spring-boot-tests/azure-spring-boot-test-application/src/main/java/com/microsoft/azure/test/Application.java  2
- azure-spring-boot-tests/azure-spring-boot-test-cosmosdb/src/test/java/com/microsoft/azure/test/cosmosdb/CosmosDBIT.java  1
- azure-spring-boot-tests/azure-spring-boot-test-core/src/main/java/com/microsoft/azure/utils/SSHShell.java  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/Microsoft/azure-spring-boot/pullRequest?prid=4450600&bid=15044855)

**Pull request merged**